### PR TITLE
Add interfaces for retrieving separate linkable downstream binaries

### PR DESCRIPTION
**Pull request by cheneym2**

### CODE REVIEW

File: source/slang-record-replay/record/slang-module.cpp [hunk 1]
```diff
@@ -213,17 +213,6 @@ namespace SlangRecord
         return res;
     }
 
-    SLANG_NO_THROW SlangResult ModuleRecorder::precompileForTarget(
```
Comment (csyonghe) [inline source/slang-record-replay/record/slang-module.cpp:216]: We must provide implementation for record-replay.

**Review by csyonghe — COMMENTED** [review]
For any compilation interface added to the slang.h header, we should implement the record/replay logic for the interface.

**Comment by csyonghe** [comment]
The new interface needs to be marked as experimental, probably name it `IModulePrecompileService_` and clearly state it as experimental and subject to removal or change in the comment.

File: include/slang.h [hunk 2]
```diff
@@ -5574,13 +5574,34 @@ namespace slang
             SlangInt32 index) = 0;
 
         virtual SLANG_NO_THROW DeclReflection* SLANG_MCALL getModuleReflection() = 0;
+    };
+    
+    #define SLANG_UUID_IModule IModule::getTypeGuid()
+
+    /* Experimental interface for doing target precompilation of slang modules */
```
Comment (cheneym2) [inline include/slang.h:5581]: Already marked experimental

**Comment by cheneym2** [comment]
> The new interface needs to be marked as experimental, probably name it `IModulePrecompileService_` and clearly state it as experimental and subject to removal or change in the comment.

I did already comment it as Experimental in slang.h.  I can additionally rename it.

**Pull request merged**